File: src/main/java/net/aspw/nightx/features/module/modules/movement/speeds/watchdog/WatchdogNew.java
package net.aspw.nightx.features.module.modules.movement.speeds.watchdog;

import net.aspw.nightx.NightX;
import net.aspw.nightx.event.EventState;
import net.aspw.nightx.event.MotionEvent;
import net.aspw.nightx.event.MoveEvent;
import net.aspw.nightx.features.module.modules.movement.Speed;
import net.aspw.nightx.features.module.modules.movement.speeds.SpeedMode;
import net.aspw.nightx.utils.MovementUtils;
import net.aspw.nightx.utils.timer.TickTimer;
import net.minecraft.potion.Potion;

public class WatchdogNew extends SpeedMode {

    private final TickTimer tickTimer = new TickTimer();
    private int groundTick = 0;

    public WatchdogNew() {
        super("WatchdogNew");
    }

    @Override
    public void onMotion(MotionEvent eventMotion) {
        final Speed speed = NightX.moduleManager.getModule(Speed.class);

        if (speed == null || eventMotion.getEventState() != EventState.PRE)
            return;

        if (!mc.thePlayer.onGround || !MovementUtils.isMoving()) {
            mc.timer.timerSpeed = 1;
        }

        if (MovementUtils.isMoving()) {
            if (mc.thePlayer.onGround) {
                if (groundTick >= 0) {
                    mc.timer.timerSpeed = 1.02f;
                    if (speed.sendJumpValue.get()) {
                        mc.thePlayer.jump();
                    } else {
                        mc.thePlayer.motionY = 0.42;
                    }
                    if (!mc.thePlayer.isPotionActive(Potion.moveSpeed)) {
                        MovementUtils.strafe(0.43f);
                    }
                    if (mc.thePlayer.isPotionActive(Potion.moveSpeed)) {
                        MovementUtils.strafe(0.63f);
                    }
                }
                groundTick++;
            } else {
                groundTick = 0;
                mc.thePlayer.motionY += -0.03 * 0.03;
            }
        }
    }

    @Override
    public void onEnable() {
        final Speed speed = NightX.moduleManager.getModule(Speed.class);

        if (speed == null)
            return;

        super.onEnable();
    }

    @Override
    public void onDisable() {
        mc.timer.timerSpeed = 1;
        super.onDisable();
    }

    @Override
    public void onUpdate() {

    }

    @Override
    public void onMotion() {

    }

    @Override
    public void onMove(MoveEvent event) {
    }
}

File: src/main/java/net/aspw/client/features/module/modules/movement/speeds/watchdog/WatchdogNew.kt
package net.aspw.client.features.module.modules.movement.speeds.watchdog

import net.aspw.client.Client
import net.aspw.client.event.EventState
import net.aspw.client.event.JumpEvent
import net.aspw.client.event.MotionEvent
import net.aspw.client.event.MoveEvent
import net.aspw.client.features.module.modules.movement.Speed
import net.aspw.client.features.module.modules.movement.speeds.SpeedMode
import net.aspw.client.utils.MovementUtils
import net.minecraft.potion.Potion

class WatchdogNew : SpeedMode("WatchdogNew") {
    private var groundTick = 0
    private var ticks = 0
    override fun onJump(event: JumpEvent) {
        if (mc.thePlayer != null && MovementUtils.isMoving())
            event.cancelEvent()
    }

    override fun onMotion(eventMotion: MotionEvent) {
        if (mc.thePlayer.hurtTime > 6) {
            mc.thePlayer.jumpMovementFactor = 0.04f
        }
        ticks++
        val speed = Client.moduleManager.getModule(
            Speed::class.java
        )
        if (speed == null || eventMotion.eventState !== EventState.PRE || mc.thePlayer.isInWater) return
        if (MovementUtils.isMoving()) {
            groundTick++
            if (mc.thePlayer.onGround) {
                mc.timer.timerSpeed = 1.01f
                if (groundTick >= 0) {
                    mc.thePlayer.motionY = 0.41999998688698
                }
                if (mc.thePlayer.isPotionActive(Potion.moveSpeed)) {
                    MovementUtils.strafe(0.578f)
                } else {
                    MovementUtils.strafe(0.428f)
                }
            } else if (ticks > 3 && mc.thePlayer.motionY > 0) {
                mc.thePlayer.jumpMovementFactor = 0.022f
            }
        } else {
            groundTick = 0
        }
    }

    override fun onEnable() {
        Client.moduleManager.getModule(
            Speed::class.java
        ) ?: return
        super.onEnable()
    }

    override fun onUpdate() {}
    override fun onMotion() {}
    override fun onMove(event: MoveEvent) {}
}